libs/portals/admin/delegation-admin/src/components/DelegationList.tsx

```diff
@@ -3,43 +3,67 @@ import { Box, Stack } from '@island.is/island-ui/core'
 import { AccessCard } from '@island.is/portals/shared-modules/delegations'
 import { useDeleteCustomDelegationAdminMutation } from '../screens/DelegationAdminDetails/DelegationAdmin.generated'
 import { useRevalidator } from 'react-router-dom'
+import React, { useState } from 'react'
+import { DelegationDeleteModal } from './DelegationDeleteModal'
 
 interface DelegationProps {
   direction: 'incoming' | 'outgoing'
   delegationsList: AuthCustomDelegation[]
 }
 
 const DelegationList = ({ delegationsList, direction }: DelegationProps) => {
-  const [deleteCustomDelegationAdminMutation] =
+  const [deleteCustomDelegationAdminMutation, { loading }] =
     useDeleteCustomDelegationAdminMutation()
   const { revalidate } = useRevalidator()
+  const [delegationToDelete, setDelegationToDelete] =
+    useState<AuthCustomDelegation | null>(null)
+
+  const deleteHandler = async (id: string) => {
+    const { data } = await deleteCustomDelegationAdminMutation({
+      variables: {
+        id,
+      },
+    })
+    if (data) {
+      revalidate()
+      setDelegationToDelete(null)
+    }
+  }
 
   return (
-    <Box marginTop={2}>
-      <Stack space={3}>
-        {delegationsList.map((delegation) => {
-          return (
-            <AccessCard
-              canModify={!!delegation.referenceId} // only allow deletion of paper delegations
-              direction={direction}
-              key={delegation.id}
-              delegation={delegation}
-              isAdminView={true}
-              onDelete={async () => {
-                const { data } = await deleteCustomDelegationAdminMutation({
-                  variables: {
-                    id: delegation.id as string,
-                  },
-                })
-                if (data) {
-                  revalidate()
+    <>
+      <Box marginTop={2}>
+        <Stack space={3}>
+          {delegationsList.map((delegation) => {
+            return (
+              <AccessCard
```

Comment:
We should reuse the `DelegationViewModal` in the admin portal to view the details of the delegation in the `onView` action of the card.